app/assets/javascripts/components/common/confirm.jsx

```diff
@@ -1,29 +1,63 @@
 import React from 'react';
 import Modal from './modal.jsx';
+import TextInput from './text_input.jsx';
 
 const Confirm = React.createClass({
   displayName: 'Confirm',
 
   propTypes: {
     onConfirm: React.PropTypes.func.isRequired,
     onCancel: React.PropTypes.func.isRequired,
-    message: React.PropTypes.string.isRequired
+    message: React.PropTypes.string.isRequired,
+    showInput: React.PropTypes.bool,
+    passcode: React.PropTypes.string,
+    joinDetails: React.PropTypes.string
+  },
+
+  getInitialState() {
+    return { passcode: '' };
   },
 
   onConfirm() {
-    this.props.onConfirm();
+    this.props.onConfirm(this.props.passcode);
   },
 
   onCancel() {
     this.props.onCancel();
   },
 
+  onChange(_valueKey, value) {
+    this.setState({ passcode: value });
```

Comment:
We can call this userInput instead of passcode, so that it works as a general prompt.